deepchecks/nlp/context.py

```diff
@@ -90,9 +90,9 @@ def __init__(self,
 
         if train is not None and test is not None:
             # check if datasets have same indexes
-            if set(train.index) & set(test.index):
-                train.reindex(list(map(lambda x: f'train-{x}', list(train.index))))
-                test.reindex(list(map(lambda x: f'test-{x}', list(test.index))))
+            if set(train.get_original_text_indexes()) & set(test.get_original_text_indexes()):
+                train._original_text_index = np.asarray([f'train-{i}' for i in train.get_original_text_indexes()])
+                test._original_text_index = np.asarray([f'test-{i}' for i in test.get_original_text_indexes()])
```

Comment:
So, if index is now an internal thingy, and is ALWAYS range(n), this will happen all the time... 
We can just change the "name" of the dataset instead or something.

Reply:
Also also, there was the "reindex" function... Why change it to accessing a private attr of a class?

Reply:
To avoid user accesable functions regarding index

Reply:
👍